src/Microsoft.DotNet.Configurer/NuGetCacheSentinel.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using Microsoft.DotNet.ProjectModel.Resolution;
+
+namespace Microsoft.DotNet.Configurer
+{
+    public class NuGetCacheSentinel : INuGetCacheSentinel
+    {
+        public static readonly string SENTINEL = $"{Product.Version}.dotnetSentinel";
+        public static readonly string INPROGRESS_SENTINEL = $"{Product.Version}.inprogress.dotnetSentinel";
```

Comment:
Talked offline about this, but I think there's potential value in having a non-versioned INPROGRESS_SENTINEL, in avoiding a set of untested conditions altogether.